### ISSUE TITLE
Android studio can’t open docs in external browser b/c of FHSenv

### ISSUE THREAD

**Issue by Profpatsch** [NixOS/nixpkgs#29513]
Android Studio uses an FHS env to make binaries work that are downloaded after the fact.
This also means that

1) `PATH` is crippled
2) The X server is not visible.
3) Chromium cannot access it’s suid wrappers when executed and crashes. This can be validated from the terminal inside Android Studio.

**Comment by primeos** [comment]
Hey @Profpatsch,

> 1. `PATH` is crippled

Yes, as you said `buildFHSUserEnv` will set it via `export PATH='/run/wrappers/bin:/usr/bin:/usr/sbin'` which might be undesirable, especially when using the terminal of Android Studio. Was this just a note or should we change that? We could e.g. append `/home/xyz/.nix-profile/bin` (`$HOME` is set) and/or `/run/current-system/sw/bin` but AFAIK we can't keep the original `$PATH` without modifying `buildFHSUserEnv`.

> 2. The X server is not visible.

What do you mean by that? `$DISPLAY` and `$XAUTHORITY` should be set and you should therefore be able to start any X11 program (e.g. `/run/current-system/sw/bin/firefox`).

> 3. Chromium cannot access it’s suid wrappers when executed and crashes. This can be validated from the terminal inside Android Studio.

That's actually very interesting...

The Chromium script tries to find the sandbox binary conditionally, e.g.:

```bash
if [ -x "/run/wrappers/bin/__chromium-suid-sandbox" ]
then
  export CHROME_DEVEL_SANDBOX="/run/wrappers/bin/__chromium-suid-sandbox"
else
  export CHROME_DEVEL_SANDBOX="/nix/store/99zn4wyy3q4a8cfhjjs42b40157vb9kp-chromium-60.0.3112.90-sandbox/bin/__chromium-suid-sandbox"
fi
```

And to be honest I've never really thought about the Chromium sandbox which means I've never enabled `security.chromiumSuidSandbox.enable` and `/run/wrappers/bin/__chromium-suid-sandbox` is therefore obviously not present at my system. And of course the binary in the nix-store can't have the suid bit set. But interestingly Chromium launches without any problems outside the FHS chroot environment while it crashes with the following message when executed from the terminal inside Android Studio:
```
[2099:2099:0918/134936.122174:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /nix/store/iy2kvj4sq30a89lc15bi7i4s3a77v5c1-chromium-60.0.3112.90-sandbox/bin/__chromium-suid-sandbox is owned by root and has mode 4755.
```
But it can't work in both cases as the mode is always `0555` the only difference is that root is not present inside the FHS env:
```bash
-r-xr-xr-x 1 root root 22784 Jan  1  1970 /nix/store/iy2kvj4sq30a89lc15bi7i4s3a77v5c1-chromium-60.0.3112.90-sandbox/bin/__chromium-suid-sandbox
-r-xr-xr-x 1 nobody nogroup 22784  1. Jan 1970  /nix/store/iy2kvj4sq30a89lc15bi7i4s3a77v5c1-chromium-60.0.3112.90-sandbox/bin/__chromium-suid-sandbox
```
Which might cause the error.

But it seems like the [SUID sandbox is being removed](https://chromium.googlesource.com/chromium/src/+/lkgr/docs/linux_suid_sandbox.md) anyway. And no matter if `security.chromiumSuidSandbox.enable` is set or not **[chrome://sandbox/](chrome://sandbox/)** will display `"You are adequately sandboxed."` and `"SUID Sandbox | No"`. I.e. this seems to be a Chromium issue and we would either need root (probably not desirable, but I've never really dealt with `buildFHSUserEnv`) or a workaround for this "bug".

**Comment by primeos** [comment]
cc @chaoflow (the Chromium part might be interesting for you)

I managed to successfully launch Chromium without the "SUID sandbox helper binary" simply by removing the code from the Bash script (i.e. even without `--disable-setuid-sandbox`):
```bash
    if [ -x "/run/wrappers/bin/${sandboxExecutableName}" ]
    then
      export CHROME_DEVEL_SANDBOX="/run/wrappers/bin/${sandboxExecutableName}"
    else
      export CHROME_DEVEL_SANDBOX="$sandbox/bin/${sandboxExecutableName}"
    fi
```
We can probably drop it then (at least `$sandbox/bin/${sandboxExecutableName}`, no sure if there might be special use-cases for `/run/wrappers/bin/${sandboxExecutableName}`).

@Profpatsch but unfortunately even without that part it doesn't work when executed from Android Studio (assuming you don't want to live dangerously :wink:):
```
[7646:7646:0918/151945.985651:FATAL:zygote_host_impl_linux.cc(107)] No usable sandbox! Update your kernel or see https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
```

**Comment by primeos** [comment]
Quick status update:

> 1) `PATH` is crippled

This is fixed now: #61839

> 2) The X server is not visible.

One can launch X11 applications (see my previous comment).

> 3) Chromium cannot access it’s suid wrappers when executed and crashes. This can be validated from the terminal inside Android Studio.

Still a problem (TODO) - I'll try to have another look at the current situation soon.

**Comment by Profpatsch** [comment]
I don’t particularly care anymore.
